blocks/bricks/__init__.py

```diff
@@ -345,15 +345,18 @@ def __init__(self, input_dim, output_dim, num_pieces, **kwargs):
         self.output_dim = output_dim
         self.num_pieces = num_pieces
 
-        self.linear_transformation = Linear(
-            name=self.name + '_linear_to_maxout', input_dim=input_dim,
-            output_dim=output_dim * num_pieces, weights_init=self.weights_init,
-            biases_init=self.biases_init, use_bias=self.use_bias)
-        self.maxout_transformation = Maxout(name=self.name + '_maxout',
-                                            num_pieces=num_pieces)
+        self.linear_transformation = Linear(name=self.name +
```

Comment:
Sorry, I got confused :)

I like what you propose for MLP. We should not create new configuration attributes when we can use those of the children. In this case configuration will effectively be propagated in the constructor. In other cases I guess we should not propagate the configuration, as this is code duplication.